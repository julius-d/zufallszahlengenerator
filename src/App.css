.Results {
  text-align: center;
  font-size: large;
}

.App-header {
  background-color: #222;
  padding: 20px;
  margin-bottom: 15px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}
