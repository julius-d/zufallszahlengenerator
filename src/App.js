import React, {Component} from 'react';
import './App.css';

class App extends Component {

  constructor() {
    super();
    this.state = {
      randomNumbers: [],
      minValue: 1,
      maxValue: 32
    }
  }

  static getRandomInt(min, max) {
    max = max + 1;
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min)) + min;
  }

  generateRandomNumbers() {
    const newRandom = App.getRandomInt(this.state.minValue, this.state.maxValue);
    this.setState({
      randomNumbers: [newRandom, ...this.state.randomNumbers]
    });
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: parseInt(value, 10)
    });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Zufallszahlengeneratort</h1>
        </header>
        <div className="App-intro">
          <div>
            <label>Min</label>
            <input name="minValue" type="number" value={this.state.minValue}
                   onChange={this.handleInputChange.bind(this)}/>
          </div>
          <div>
            <label>Max</label>
            <input name="maxValue" type="number" value={this.state.maxValue}
                   onChange={this.handleInputChange.bind(this)}/>
          </div>
          <button onClick={this.generateRandomNumbers.bind(this)}>WÃ¼rfeln</button>
          {this.state.randomNumbers.map((randomNumber, i) => <div key={i}><strong>{randomNumber}</strong></div>)}
        </div>
      </div>
    );
  }
}

export default App;
