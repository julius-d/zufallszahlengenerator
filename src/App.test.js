import React from "react";
import App from "./App";
import { mount } from "enzyme";

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("creates random numbers", () => {
    expect(wrapper.find("ResultList").props().randomNumbers.length).toEqual(0);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("ResultList").props().randomNumbers.length).toEqual(1);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("ResultList").props().randomNumbers.length).toEqual(2);
  });

  it("max value can be changed", () => {
    expect(wrapper.state().maxValue).toEqual(32);
    changeMaxValueTo("10");
    expect(wrapper.state().maxValue).toEqual(10);
  });

  it("min value can be changed", () => {
    expect(wrapper.state().minValue).toEqual(1);
    changeMinValueTo("2");
    expect(wrapper.state().minValue).toEqual(2);
  });

  it("max value is limited to 1000000000", () => {
    expect(wrapper.state().maxValue).toEqual(32);
    changeMaxValueTo("90000000009");
    expect(wrapper.state().maxValue).toEqual(1000000000);
  });

  it("max value cannot be chars", () => {
    expect(wrapper.state().maxValue).toEqual(32);
    changeMaxValueTo("foo");
    expect(wrapper.state().maxValue).toEqual("");
  });

  function enableLottoMode() {
    wrapper
      .find("input#lotto")
      .simulate("change", { target: { checked: true } });
  }

  function changeMaxValueTo(value) {
    wrapper
      .find("input#max")
      .simulate("change", { target: { value: value, name: "maxValue" } });
  }

  function changeMinValueTo(value) {
    wrapper
      .find("input#min")
      .simulate("change", { target: { value: value, name: "minValue" } });
  }

  it("changes button name", () => {
    expect(wrapper.state().lotto).toEqual(false);
    expect(wrapper.find("button").text()).toEqual("WÃ¼rfeln");
    enableLottoMode();

    expect(wrapper.state().lotto).toEqual(true);
    expect(wrapper.find("button").text()).toEqual("Zahl ziehen");
  });

  it("creates all possible lotto numbers", () => {
    enableLottoMode();
    changeMaxValueTo("10");
    expect(wrapper.find("ResultList").props().randomNumbers.length).toEqual(0);
    for (let i = 0; i < 10; i++) {
      wrapper.find("button").simulate("click");
    }
    let randomNumbers = wrapper.find("ResultList").props().randomNumbers;

    expect(randomNumbers).toHaveLength(10);
    expect(randomNumbers).toContain(1);
    expect(randomNumbers).toContain(2);
    expect(randomNumbers).toContain(3);
    expect(randomNumbers).toContain(4);
    expect(randomNumbers).toContain(5);
    expect(randomNumbers).toContain(6);
    expect(randomNumbers).toContain(7);
    expect(randomNumbers).toContain(8);
    expect(randomNumbers).toContain(9);
    expect(randomNumbers).toContain(10);
  });

  it("does not fail on choosing one number more than possible", () => {
    enableLottoMode();
    changeMaxValueTo("2");
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
  });
});
