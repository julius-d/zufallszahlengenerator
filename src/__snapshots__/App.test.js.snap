// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Zufallszahlengenerator
    </h1>
  </header>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-6"
      >
        <div
          className="form-group form-group-lg"
        >
          <label
            className="control-label"
            htmlFor="min"
          >
            Min
          </label>
           
          <input
            className="form-control"
            id="min"
            name="minValue"
            onChange={[Function]}
            type="number"
            value={1}
          />
        </div>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="form-group form-group-lg"
        >
          <label
            className="control-label"
            htmlFor="max"
          >
            Max
          </label>
           
          <input
            className="form-control"
            id="max"
            name="maxValue"
            onChange={[Function]}
            type="number"
            value={32}
          />
        </div>
      </div>
      <div
        className="col-xs-12"
      >
        <button
          className="btn btn-lg btn-primary btn-block"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          WÃ¼rfeln
        </button>
      </div>
    </div>
  </div>
  <div
    className="Results"
  />
</div>
`;
